ci/jenkins/jobs/macros.yaml

```diff
@@ -78,14 +78,35 @@
     name: builder-workload-cluster-cleanup
     builders:
       - shell: |-
+          #!/bin/bash
           cluster="${CLUSTERNAME}"
 
           echo '=== Clean up cluster ==='
           export KUBECONFIG="/var/lib/jenkins/kubeconfig-mgmt"
-          kubectl delete Machine ${cluster}-controlplane-0 -n ${cluster} || true
-          kubectl delete MachineDeployment ${cluster}-md-0 -n ${cluster} || true
-          kubectl delete cluster ${cluster} -n ${cluster} || true
-          kubectl delete ns ${cluster} || true
+
+          kubectl delete --ignore-not-found=true Machine ${cluster}-controlplane-0 -n ${cluster}
+          kubectl delete --ignore-not-found=true MachineDeployment ${cluster}-md-0 -n ${cluster}
+
+          retry=12
+          while [ "${retry}" -gt 0 ]; do
+            echo '=== Waiting for Machines to be deleted ==='
+            machines="$(kubectl get machine -n ${cluster} --no-headers=true 2>/dev/null | wc -l)"
+            if [ "${machines}" -eq 0 ]; then
+              break
+            fi
+            echo '=== Remaining Machines ==='
+            kubectl get machine -n ${cluster}
+            sleep 10
+            retry=$((retry-1))
+          done
+          if [ "${retry}" -eq 0 ]; then
+            echo "=== Failed to delete machines! ==="
+            exit 1
+          fi
+
+          kubectl delete --ignore-not-found=true vspherecluster ${cluster} -n ${cluster}
```

Comment:
Let's make it consistent.
* vspherecluster  -> VSphereCluster
* cluster -> Cluster